### PR TITLE
Fix vulnerable (transitive) dependencies

### PR DESCRIPTION
**Pull request by MagielBruntink**
## Description
FASTEN dependencies have several known vulnerabilities in them. This PR fixes those. 

See the attached CSV file for the list:
[fasten-all-vulnerabilities.csv](https://github.com/fasten-project/fasten/files/7711021/fasten-all-vulnerabilities.csv)

## Motivation and context
Vulnerabilties are bad.

## Testing
Ran unit tests successfully.

## Task list  
- [ ] Validate whether all vulnerabilities are resolved. With FASTEN itself :-)
- [x] Succesfull CI run.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,33 +88,53 @@
         <module>server</module>
     </modules>
 
+    <dependencyManagement>
+        <dependencies>
```
Comment (proksch) [inline pom.xml:92]: Why were these additional dependencies added?

Comment (mir-am) [reply pom.xml:92]: And why here?
We add these common dependencies (like `log4j`) to the server module so that all the plug-ins can use them.

**Review by mir-am — APPROVED** [review]
Thanks for updating the dependencies. Overall, LGTM. I have just one minor comment.

File: pom.xml (hunk 1 above)
Comment (MagielBruntink) [reply pom.xml:92]: The depedencyManagement section controls versions of (transitive) dependencies used. These few are needed here to upgrade vulnerable dependencies of shrinkwrap (commons-compress, jsoup, log4j-api). If the dependencyManagement section is added to the POM of fasten/core it doesn't control the versions of modules under fasten/analyzer, annoyingly. So it needs to go here, in the parent POM.

Comment (mir-am) [reply pom.xml:92]: I see!

**Comment by MagielBruntink** [comment]
> Thanks for updating the dependencies. Overall, LGTM. I have just one minor comment.

I don't see log4j in the server/pom.xml. It's down somewhere else as a transitive dep.

File: pom.xml (hunk 1 above)
Comment (proksch) [reply pom.xml:92]: Oh, I was too fast in skimming the changes and missed this detail...

**Comment by MagielBruntink** [comment]
We could consider a more elaborate updating of dependencies in this PR. I saw several that could use a new version, but without a vulnerability present. Also there is a lot of duplication in the inclusions of deps all over the place. Do we want to make an effort here to clean that up?

**Comment by proksch** [comment]
I think this would be a great idea. I would just postpone this until we have finalized the updates to the build infrastructure that we are currently discussing in #358 .

**Pull request merged**